.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,21 @@
+name: Checkstyle
```

Comment:
Is this workflow needed?

Reply:
I had created a separate workflow because I was thinking more along the lines of separation of interests, even though there was some code duplication.... But I'll keep it in the maven.yml after all since that is more efficient.